modules/gmc-agent/src/main/java/org/gridgain/action/query/CursorHolder.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.action.query;
+
+import org.apache.ignite.cache.query.FieldsQueryCursor;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Cursor holder.
+ */
+public class CursorHolder implements AutoCloseable {
```

Comment:
Why we need CursorHolder if we already has QueryHolder?
Why we have two keys qryId + cursorId instead of qryId?

Reply:
We want to full support of a multi-statement queries, because QueryHolder has set of CursorHolder's.
We need cursor id for getting next page result for the specific query statement.